From d3950f5060e70f01fc7fb4ab99afaf03543e4204 Mon Sep 17 00:00:00 2001
From: Gurumoorthy Santhakumar <quic_gsanthak@quicinc.com>
Date: Mon, 2 Dec 2024 11:20:18 +0530
Subject: [PATCH] driver: firmware: scm: Clear unnecessary flag in private data

An unnecessary flag was set in the private data, causing the
abnormal magic bit to be set. This led to continuous resets
at u-boot upon a kernel crash. This commit clears the flag
to prevent this issue.

Change-Id: Ic23b325e5c3f35b58790c313ed5d2f68b5740b8b
Signed-off-by: Gurumoorthy Santhakumar <quic_gsanthak@quicinc.com>
(cherry picked from commit d3d2a46c309cb509a596b3bcd058ccb2af54087a)
---
 drivers/firmware/qcom_scm.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/firmware/qcom_scm.c b/drivers/firmware/qcom_scm.c
index 9272109ba279..f6fee5431325 100644
--- a/drivers/firmware/qcom_scm.c
+++ b/drivers/firmware/qcom_scm.c
@@ -3541,7 +3541,6 @@ struct scm_priv_data ipq9574_data = {
 };
 
 struct scm_priv_data ipq5424_data = {
-	.flag = (SDI_DISABLE | ABNORMAL_MAGIC),
 	.milestone_mask = BIT(12)
 };
 
-- 
2.34.1

